packages/rule-typescript-config/src/target.ts

```diff
@@ -0,0 +1,235 @@
+        },
+        id: 'typescript-config/target',
+        schema: [],
+        scope: RuleScope.local
+    }
+
+    public constructor(context: RuleContext) {
+        const Targets: Map<string, string> = new Map([
+            ['es3', 'es3'],
+            ['es5', 'es5'],
+            ['es6', 'es2015'],
+            ['es2015', 'es2015'],
+            ['es2016', 'es2016'],
+            ['es2017', 'es2017'],
+            ['esnext', 'esnext']
+        ]);
+
+        /**
+         * ES compatibility matrix. For each version of ES, it enumerates the minimum browser with
+         * acceptable support.
+         *
+         * Data obtained from:
+         * * http://kangax.github.io/compat-table/es5/
+         * * http://kangax.github.io/compat-table/es6/
+         * * http://kangax.github.io/compat-table/es2016plus/
+         */
+        const compatMatrix = {
+            es5: {
+                chrome: 57,
+                edge: 12,
+                firefox: 45,
+                ie: 9,
+                ie_mob: 11, // eslint-disable-line camelcase
+                opera: 44,
+                safari: 8
+            },
+            es2015: { // eslint-disable-line sort-keys
+                chrome: 57,
+                edge: 13,
+                firefox: 45,
+                ie: 'none',
+                ie_mob: 'none', // eslint-disable-line camelcase
+                opera: 44,
+                safari: 10
+            },
+            es2016: {
+                chrome: 57,
+                edge: 14,
+                firefox: 52,
+                ie: 'none',
+                ie_mob: 'none', // eslint-disable-line camelcase
+                opera: 44,
+                safari: 10.1
+            },
+            es2017: {
+                chrome: 58,
+                edge: 16,
+                firefox: 53,
+                ie: 'none',
+                ie_mob: 'none', // eslint-disable-line camelcase
+                opera: 45,
+                safari: 10.1
+            }
+        };
+
+        const getMajor = (version: string | number): number => {
+            if (typeof version === 'number') {
+                return version;
+            }
+
+            // Something like "4.4.3-4.4.4" --> 4
+            return parseInt(version.split('-')[0].split('.')[0]);
+        };
+
+        /**
+         * Checks if `version1` is oder (i.e.: lower version) than `version2` taking into
+         * account ranges and minor versions. E.g.: `4.4.3-4.4.4` will use `4.4.3` to compare
+         */
+        const isOlder = (version1: number | string, version2: number | string): boolean => {
+
+            if (typeof version1 === 'number' && typeof version2 === 'number') {
+                return version1 < version2;
+            }
+
+            // Can't store `typeof` or otherwise TypeScript complains about `includes` not being on `number`
+            if (typeof version1 === 'string' && typeof version2 === 'string') {
```

Comment:
IMO, this is to complex (or I'm missing something). Why don't you just parse the versions?
```
if(typeof v1 === 'string') {
v1 = getMajor(v1)
}
if(typeof v2 === 'string') {
v2 = getMajor(v2)
}
return isOlder(v1, v2)
```

Reply:
Browser versions are complicated...
We could be comparing:
* `safari 10.1` and `safari 10.0`
* `android 3.0.1` and `android 4.4.3`
* In some cases we get a range as a version: `android 4.4.3-4.4.4`.

I should put some comments or try to refactor the code to make it clearer.

Reply:
but at the end, you just compare the major version, isn't it?

Reply:
I should compare the mayor only if the other tests have failed.

Reply:
ah, ok